### ISSUE TITLE
still after training getting this error, I'm still learning to speak with humans. What you mean?

### ISSUE THREAD

**Issue by saurabhbidwai** [samtecspg/articulate#279]
### Issues
> *If you're having trouble with Articulate we definitely want to help you. If you can provide a little bit of information it will make our troubleshooting job easier.*

Please describe the issue you are having here and answer the question below.

- Did you download a release or pull from Master? Which release?
0.9.4
- What operating system are you running?
windows 10
- Are you running locally or in the cloud?
locally
- Which browser are you using?
chrome
----

### Feature Request
> *If you've got a feature request we want to hear it*

Please describe the feature you would like us to add.

If possible provide the use case for the feature and how you would use it.

**Comment by dcalvom** [comment]
Hi @saurabhbidwai, that's a fallback response, those are defined in the agent screen, you can edit those to whatever you like. If you are having a fallback response that means, that the user text wasn't recognized with a good confidence.

**Comment by saurabhbidwai** [comment]
u can see I have trained the model on some dialogues but still getting this
error

**Comment by saurabhbidwai** [comment]
![2018-07-20](https://user-images.githubusercontent.com/22415674/42995114-a7a94cb4-8c2d-11e8-9fda-06340c93dbe4.png)

**Comment by dcalvom** [comment]
In this case you need to "tune" your agent. Please go to `http://localhost:7500/documentation` and test the endpoint `/agent/{id}/parse` with the text `ham pizza` or `I want pizza` and check the confidence.

Once you test several examples you can have an idea of what threshold to use for the domain recognition threshold (for the agent) and for the intent recognition threshold (per domain).

A quick explanation. When you have an agent with multiple domains, let's say:

1. Small Talk domain: for greeting, gratitude, and goodbye
2. Order domains: for order placing, order review, etc...
3. Tracking domain: for track your order or get other order status info

Your agent will create a domain recognizer. Therefore, you have to define the threshold to define if a text belongs to a domain or it is unrecognized and the agent should use a fallback (your case).

Once the domain is defined, then, you are going to use the intent recognition threshold to define if the intent inside the domain is with enough confidence to be answered. If not, again, the agent will use the fallback.

**Comment by saurabhbidwai** [comment]
i tunned domain recognition threshold accordingly still not able to get desired results

**Comment by wrathagom** [comment]
Can you export your agent for us to review? Likely it is just a combination of too little training data and not the right threshold. You can go to `localhost:7500/documentation` and look under agent to find the export endpoint.

**Comment by saurabhbidwai** [comment]
{
  "language": "en",
  "timezone": "UTC",
  "agentName": "Pizza Bot",
  "useWebhook": false,
  "description": "order your pizza",
  "extraTrainingData": false,
  "fallbackResponses": [
    "Sorry can you rephrase that?",
    "I'm still learning to speak with humans. What you mean?"
  ],
  "domainClassifierThreshold": 0.35,
  "entities": [
    {
      "regex": null,
      "uiColor": "#3f51b5",
      "entityName": "Toppings",
      "examples": [
        {
          "value": "Pepperoni",
          "synonyms": [
            "Pepperoni"
          ]
        },
        {
          "value": "Sausage",
          "synonyms": [
            "Sausage"
          ]
        },
        {
          "value": "Ham",
          "synonyms": [
            "Ham"
          ]
        },
        {
          "value": "Pineapple",
          "synonyms": [
            "Pineapple"
          ]
        }
      ]
    }
  ],
  "domains": [
    {
      "model": "e34eefcf-f712-a76c-cc56-5b77c5a9ac93",
      "enabled": true,
      "domainName": "Order Receiver",
      "lastTraining": "Wed Jul 18 2018 05:00:21 GMT+0000 (UTC)",
      "intentThreshold": 0.5,
      "extraTrainingData": false,
      "intents": [
        {
          "intentName": "greeting",
          "useWebhook": false,
          "examples": [
            {
              "userSays": "hola",
              "entities": []
            },
            {
              "userSays": "hey",
              "entities": []
            },
            {
              "userSays": "hello",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "hi"
            }
          ],
          "scenario": {
            "slots": [],
            "scenarioName": "greeting",
            "intentResponses": [
              "Hello, My name is Myzza, How can I help you?"
            ]
          }
        },
        {
          "intentName": "thanks",
          "useWebhook": false,
          "examples": [
            {
              "entities": [],
              "userSays": "ok"
            },
            {
              "userSays": "okay",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "excellent"
            }
          ],
          "scenario": {
            "slots": [],
            "scenarioName": "thanks",
            "intentResponses": [
              "Thank you!"
            ]
          }
        },
        {
          "intentName": "toppings",
          "useWebhook": false,
          "examples": [
            {
              "entities": [],
              "userSays": "give me pizza"
            },
            {
              "userSays": "i want pizza",
              "entities": []
            },
            {
              "userSays": "ham cheese pizza",
              "entities": [
                {
                  "end": 3,
                  "start": 0,
                  "value": "ham",
                  "entity": "Toppings"
                },
                {
                  "end": 10,
                  "start": 4,
                  "value": "cheese",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "cheese pizza",
              "entities": [
                {
                  "end": 6,
                  "value": "cheese",
                  "start": 0,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "ham pizza",
              "entities": [
                {
                  "end": 3,
                  "start": 0,
                  "value": "ham",
                  "entity": "Toppings"
                }
              ]
            }
          ],
          "scenario": {
            "scenarioName": "toppings",
            "slots": [
              {
                "entity": "Toppings",
                "isList": true,
                "slotName": "Toppings",
                "isRequired": true,
                "textPrompts": [
                  "what toppings?"
                ]
              }
            ],
            "intentResponses": [
              "{{slots.Toppings.value}} on its way!",
              "you order on its way!"
            ]
          }
        }
      ]
    }
  ]
}

**Comment by wrathagom** [comment]
@saurabhbidwai sorry for the delay I will look at this in depth tomorrow. We've been cramming for a big presentation.

**Comment by saurabhbidwai** [comment]
np @wrathagom

**Comment by wrathagom** [comment]
@saurabhbidwai I am not having any trouble with your agent.

<img width="1648" alt="2018-07-26_10-04-27" src="https://user-images.githubusercontent.com/1779894/43267407-dd3b0614-90bb-11e8-8576-20b01539c341.png">

Can you try to make a slight change to your agent (add an extra example) and re-train? Another thing to try would be to export your agent, delete it, and re-import it. Just in case something happened during training. Are there any error messages in the console?

**Comment by saurabhbidwai** [comment]
I have added some extra examples and it's working properly now. 
Thanks

**Comment by wrathagom** [comment]
Great! Let us know if you have any other troubles.

**Comment by saurabhbidwai** [comment]
hello @wrathagom , I'm trying to understand the working of the Articulate but unable find any solid documents, can u guide me for it?

**Comment by wrathagom** [comment]
Sorry @saurabhbidwai our documentation is sorely lacking. Though we've hired someone to fix it. For now there are some docs here: https://samtecspg.github.io/articulate/ and more on our blog here: https://blog.spg.ai

**Comment by wrathagom** [comment]
@saurabhbidwai let's create a new issue for this one. When you do please include the output of `docker-compose ps` as well as `docker-compose logs`
